climateconnect/climateconnect
# Fix: Prevent creation of organizations with same name

## Description
Closes #1040

Checking if name already exists changed to ignore case sensitivity i.e. if an organization uses the name "test", another organization cannot be created under the name "TEST".

Added a check when editing an organization name if that organization name is in use. If yes, then an alert will be shown and the user must choose another.

Implementation: 
Added a new view and  API call to the backend to lookup if the input matches an existing organization name or an existing translation of an organization name (basic interface). Additional checks are added for the case of creation in the detailed interface or editing.

## Additional
PR implements a change to project previews. Before displaying # of likes or comments, a check is made if either has at least 3.

## Test plan for creation (basic info step)

1. Go to create organization
2. For the name, enter a name of an existing organization and attempt to continue
3. An error should be thrown

## Test plan for creation (detailed info step)

1. Go to create organization
2. For the name, enter a new name of an organization and continue
3. You should be on the detailed organization creation interface
4. Change the name to one of an organization that is taken
5. Attempt to publish and and error should be shown

Additionally, for German page "veröffentlichen" should open the auto translate interface. Attempting to publish here with an organization name in use should throw an error.


## Test plan for editing

1. Choose an Org to edit
2. Enter the name of another org and attempt to save
3. An alert should pop up


## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)
